goldenmaster/apigear/olink/olinkconnection.cpp

```diff
@@ -233,18 +194,19 @@ void OlinkConnection::scheduleProcessMessages(long delayMiliseconds)
 
 void OlinkConnection::processMessages(Poco::Util::TimerTask& /*task*/)
 {
-    if (!m_socket && !m_disconnectRequested) {
+    if (m_socket.isClosed()) {
```

Comment:
This does not seem to produce the same behavior.
isClosed only returns true if the socket is a nullptr, it is not checked whether the user already requested a disconnect.
It is just edge case and we probably haven't defined the exact behavior.
However in this situation we will keep sending messages as long as the socket is open, regardless of the requested state